pacai/core/distanceCalculator.py

```diff
@@ -54,103 +42,58 @@ def getDistanceOnGrid(self, pos1, pos2):
-    if x == int(x):
-        return [(x, 0)]
-    return [(intX, x - intX), (intX + 1, intX + 1 - x)]
-
-##########################################
-# MACHINERY FOR COMPUTING MAZE DISTANCES #
-##########################################
-
-distanceMap = {}
-
-class DistanceCalculator:
-    def __init__(self, layout, distancer):
-        self.layout = layout
-        self.distancer = distancer
-        self.cache = {}
-
-    def run(self):
-        if self.layout.walls not in self.cache:
-            self.cache[self.layout.walls] = computeDistances(self.layout)
-
-        self.distancer._distances = self.cache[self.layout.walls]
-
-def computeDistances(layout):
-    """
-    Runs UCS to all other positions from each position.
-    """
-
-    distances = {}
-    allNodes = layout.walls.asList(False)
-
-    for source in allNodes:
-        dist = {}
-        closed = {}
-
-        for node in allNodes:
-            dist[node] = sys.maxsize
-
-        queue = priorityQueue.PriorityQueue()
-        queue.push(source, 0)
-        dist[source] = 0
+    def round2D(self, pos):
+        intX = int(pos[0])
+        intY = int(pos[1])
+        deltaX = float(abs(pos[0] - intX))
+        deltaY = float(abs(pos[1] - intY))
 
-        while not queue.isEmpty():
-            node = queue.pop()
-            if node in closed:
-                continue
+        if deltaX >= 0.5 and deltaY >= 0.5:
+            return (intX + 1, intY + 1)
+        elif deltaX < 0.5 and deltaY >= 0.5:
+            return (intX, intY + 1)
+        elif deltaX >= 0.5 and deltaY < 0.5:
+            return (intX + 1, intY)
+        else:
+            return (intX, intY)
 
-            closed[node] = True
-            nodeDist = dist[node]
-            adjacent = []
-            x, y = node
+    def computeDistances(self, layout):
 
-            if not layout.isWall((x, y + 1)):
-                adjacent.append((x, y + 1))
+        distances = {}
+        allNodes = layout.walls.asList(False)
 
-            if not layout.isWall((x, y - 1)):
-                adjacent.append((x, y - 1))
+        for source in allNodes:
+            dist = {}
+            closed = []
 
-            if not layout.isWall((x + 1, y)):
-                adjacent.append((x + 1, y))
+            for node in allNodes:
+                dist[node] = DEFAULT_DISTANCE
 
-            if not layout.isWall((x - 1, y)):
-                adjacent.append((x - 1, y))
+            adjacent = [source]
+            dist[source] = 0
 
-            for other in adjacent:
-                if other not in dist:
+            while len(adjacent) != 0:
+                node = adjacent.pop(0)
+                if node in closed:
                     continue
+                closed += [node]
+                x, y = node
 
-                oldDist = dist[other]
-                newDist = nodeDist + 1
-                if newDist < oldDist:
-                    dist[other] = newDist
-                    queue.push(other, newDist)
```

Comment:
We have a function for this: Actions.getLegalNeighbors(prevPosition, layout.walls)